### PR TITLE
Add subpass parameter type to materials

### PR DESCRIPTION
**Pull request by bejado**

### CODE REVIEW

File: tools/matc/src/matc/ParametersProcessor.cpp [hunk 1]
```diff
@@ -219,7 +243,7 @@ static bool processParameters(MaterialBuilder& builder, const JsonishValue& v) {
     bool ok = true;
     for (auto value : jsonArray->getElements()) {
         if (value->getType() == JsonishValue::Type::OBJECT) {
-            ok |= processParameter(builder, *value->toJsonObject());
```
Comment (romainguy) [inline tools/matc/src/matc/ParametersProcessor.cpp:222]: Damn booleans :)

**Review by romainguy — APPROVED** [review]
Does matinfo need to be updated too?

**Comment by bejado** [comment]
> Does matinfo need to be updated too?

matinfo will work, although it won't report the subpass as a parameter. In order to do so we'd have to store the subpass in a new material chunk, which I thought was a bit overkill for now.

File: libs/filamat/src/shaders/CodeGenerator.cpp [hunk 2]
```diff
@@ -339,6 +339,29 @@ io::sstream& CodeGenerator::generateSamplers(
...
+        SubpassInfo subpass) const {
+    if (!subpass.isValid) {
+        return out;
+    }
+
+    CString subpassName =
+            SamplerInterfaceBlock::getUniformName("MaterialParams", subpass.name.c_str());
+
+    char const* const typeName = "subpassInput";
+    // In our Vulkan backend, subpass inputs always live in descriptor set 2. (ignored for GLES)
```
Comment (prideout) [inline libs/filamat/src/shaders/CodeGenerator.cpp:352]: 👍

**Review by prideout — APPROVED** [review]

**Comment by romainguy** [comment]
Yeah I agree it's overkill but we already list the parameters and their types in a chunk. Could we just reuse the existing mechanism for subpass input?

**Comment by bejado** [comment]
> Yeah I agree it's overkill but we already list the parameters and their types in a chunk. Could we just reuse the existing mechanism for subpass input?

It's difficult to reuse the existing mechanism, but I looked into it and I can just add a new chunk type for subpasses.

**Comment by romainguy** [comment]
The Material API lets you query all the declared parameters, it's not just for matinfo, it can be used from Filament directly. We need to make sure this query API works at the very least.

**Comment by bejado** [comment]
> The Material API lets you query all the declared parameters, it's not just for matinfo, it can be used from Filament directly. We need to make sure this query API works at the very least.

Agreed. I'll put that change up in a separate PR to make it more straightforward to review.

**Pull request merged**